collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java

```diff
@@ -2647,7 +2649,7 @@ public void savingComplete(SaveResult saveResult) {
                 } else {
                     formController.getTimerLogger().logTimerEvent(TimerLogger.EventTypes.FORM_EXIT, 0, null, false, true);         // Force writing of audit since we are exiting
                 }
-                enqueueFinalizedForm();
+                enqueueAutosendJob();
```

Comment:
This doesn't always enqueue so `enqueueAutosendJobIfNeeded` might be more accurate. Another option is to pull out the enqueuing condition so it's really clear when jobs are and aren't enqueued.

Reply:
Thanks. that makes sense 😄

Reply:
I'm going to make a little PR checklist of some of these highlights for my personal reviews. Because at first addng `IfNeeded` didn't come to mind since previously the `AutoSendWorker` was being enqueued no matter what but with the changes in logic comes changes to the method's naming as well 😄

Reply:
> with the changes in logic comes changes to the method's naming as well

Yes, absolutely. It may seem nit-picky but it's the kind of thing that can cause a lot of confusion in the future!